### DISCUSSION TITLE
regular certificate issue for Kuwait channels

### DISCUSSION THREAD

**Discussion by andaluspc** [streamlink/streamlink#5570] [Q&A] [answered]
### Description

Hello team. hope anyone may have an idea about how this issue can be mitigated. I run a script to get me stream URL every some hours. not sure if this is the cause of the issue somehow.

While trying to get stream URL for Kuwait channels, I regularly get stuck in getting certificate error (which I found to be related to URL in SSL cert doesn't match the actual URL hostname).
command I use is the following (and, if I disable ssl verification, gives me a streamlink. However, streaml ULR I get doesn't work on iptv players. on VLC, it keeps getting me certificate warning which, if I accept, get the stream to be played.

The issue found only when I am on my home connection (I don't do extensive test except at home. so, no clue if issue can be replicated otherwise). The following are some more info about the nature of the problem;
- When issue is happening, I can't even load the stream on the web page itself! (so, issue isn't only related to streamlink)
- Sometimes (not always), issue disappears if I get a new public IP on my home router.
- When I have the problem, I connect to VPN and channels start to work. When I disconnect VPN it stops working again. (issue is related to my source IP obviously).
- While trying proxy-option with streamlink, I couldn't successfully get stream url (picked up proxies from https://www.sslproxies.org/)
- Using online free http proxy: I got it to work with https://www.croxyproxy.com/ but it DID NOT work  with https://hide.me/proxy)

Below is debug output. here is also latest stream URL I generated (Let me know if you can play it without any problem. it is valid for 24 hours):
Desktop:~$ streamlink --stream-url --http-no-ssl-verify https://www.media.gov.kw/LiveTV.aspx?PanChannel=KTVEthraa
https://hiplayer.hibridcdn.net/p/kwmedia-kwtvethraa?p=53&s=1695627634&e=1695714034&cf=1695714034&h=21c1e3a667d20c041638a34bed6416a4

Desktop:~$ streamlink --http-no-ssl-verify https://www.media.gov.kw/LiveTV.aspx?PanChannel=KTVEthraa --loglevel=debug
[cli][debug] OS:         Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
[cli][debug] Python:     3.10.12
[cli][debug] OpenSSL:    OpenSSL 3.0.2 15 Mar 2022
[cli][debug] Streamlink: 6.2.0
[cli][debug] Dependencies:
[cli][debug]  certifi: 2023.7.22
[cli][debug]  isodate: 0.6.1
[cli][debug]  lxml: 4.9.3
[cli][debug]  pycountry: 22.3.5
[cli][debug]  pycryptodome: 3.19.0
[cli][debug]  PySocks: 1.7.1
[cli][debug]  requests: 2.31.0
[cli][debug]  trio: 0.22.2
[cli][debug]  trio-websocket: 0.10.4
[cli][debug]  typing-extensions: 4.8.0
[cli][debug]  urllib3: 2.0.4
[cli][debug]  websocket-client: 1.6.3
[cli][debug] Arguments:
[cli][debug]  url=https://www.media.gov.kw/LiveTV.aspx?PanChannel=KTVEthraa
[cli][debug]  --loglevel=debug
[cli][debug]  --http-no-ssl-verify=False
[cli][info] Found matching plugin hiplayer for URL https://www.media.gov.kw/LiveTV.aspx?PanChannel=KTVEthraa
[warnings][insecurerequestwarning] Unverified HTTPS request is being made to host 'www.media.gov.kw'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  /home/andalus/.local/lib/python3.10/site-packages/urllib3/connectionpool.py:1095
[plugins.hiplayer][debug] JS URL=https://hiplayer.hibridcdn.net/l/kwmedia-kwtvethraa
[warnings][insecurerequestwarning] Unverified HTTPS request is being made to host 'hiplayer.hibridcdn.net'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  /home/andalus/.local/lib/python3.10/site-packages/urllib3/connectionpool.py:1095
[utils.l10n][debug] Language code: en_US
[warnings][insecurerequestwarning] Unverified HTTPS request is being made to host 'hiplayer.hibridcdn.net'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  /home/andalus/.local/lib/python3.10/site-packages/urllib3/connectionpool.py:1095
Available streams: 240p (worst), 360p, 480p, 720p, 1080p (best)

**Comment by bastimeyer** [comment]
Streamlink already has a plugin for this site, so there shouldn't be a need to find stream URLs manually, unless something's broken, but in this case, a plugin issue should be reported:  
https://streamlink.github.io/plugins.html#hiplayer

----

I don't see any TLS errors though, neither on `*.media.gov.kw`, nor on `kwmedia.hibridcdn.net` (HLS streams are hosted here). The certs are good and the servers are properly configured, apart from accepting old/deprecated TLS versions and ciphers.

- https://www.ssllabs.com/ssltest/analyze.html?d=www.media.gov.kw&s=62.150.155.212
- https://www.ssllabs.com/ssltest/analyze.html?d=www.media.gov.kw&s=91.140.227.212
- https://www.ssllabs.com/ssltest/analyze.html?d=kwmedia.hibridcdn.net&latest

> [cli][debug] OpenSSL: OpenSSL 3.0.2 15 Mar 2022
> certifi: 2023.7.22

Try updating your Python environment to the latest version, including OpenSSL, and make sure that your system clock is properly synced.

Streamlink uses the CA bundle provided by [`certifi`](https://pypi.org/project/certifi/), which you're using the latest version of.

When the `--http-no-ssl-verify` CLI option is set, then the `Session.verify` attribute gets set to `False`, which makes `requests` tell `urllib3` to disable TLS verification on HTTPS connections:  
- https://requests.readthedocs.io/en/latest/api/#requests.Session.verify
- https://github.com/psf/requests/blob/v2.31.0/requests/adapters.py#L249-L275
- https://urllib3.readthedocs.io/en/2.0.4/reference/urllib3.connection.html#urllib3.connection.HTTPSConnection

----

TLS issues are not a problem of Streamlink though, so you should look for help somewhere else. There's not much that can be done here.

I have no idea why using a VPN or proxy would cause different results. The client IP address shouldn't matter when validating TLS server certificates. "I regularly get stuck in getting certificate error" doesn't describe the **actual** error message you've encountered, so nobody can help you.

You can use [`openssl s_client`](https://www.openssl.org/docs/man3.1/man1/openssl-s_client.html) to debug TLS issues on your system.

```sh
echo | openssl s_client \
  -verifyCAfile /home/andalus/.local/lib/python3.10/site-packages/certifi/cacert.pem \
  -connect media.gov.kw:443
```

----

```
$ echo | openssl s_client -verifyCAfile ~/venv/streamlink-311/lib/python3.11/site-packages/certifi/cacert.pem -connect media.gov.kw:443
CONNECTED(00000003)
depth=2 C = US, O = Amazon, CN = Amazon Root CA 1
verify return:1
depth=1 C = US, O = Amazon, CN = Amazon RSA 2048 M01
verify return:1
depth=0 CN = media.gov.kw
verify return:1
---
Certificate chain
 0 s:CN = media.gov.kw
   i:C = US, O = Amazon, CN = Amazon RSA 2048 M01
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Mar 31 00:00:00 2023 GMT; NotAfter: Apr 28 23:59:59 2024 GMT
 1 s:C = US, O = Amazon, CN = Amazon RSA 2048 M01
   i:C = US, O = Amazon, CN = Amazon Root CA 1
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Aug 23 22:21:28 2022 GMT; NotAfter: Aug 23 22:21:28 2030 GMT
 2 s:C = US, O = Amazon, CN = Amazon Root CA 1
   i:C = US, ST = Arizona, L = Scottsdale, O = "Starfield Technologies, Inc.", CN = Starfield Services Root Certificate Authority - G2
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: May 25 12:00:00 2015 GMT; NotAfter: Dec 31 01:00:00 2037 GMT
 3 s:C = US, ST = Arizona, L = Scottsdale, O = "Starfield Technologies, Inc.", CN = Starfield Services Root Certificate Authority - G2
   i:C = US, O = "Starfield Technologies, Inc.", OU = Starfield Class 2 Certification Authority
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Sep  2 00:00:00 2009 GMT; NotAfter: Jun 28 17:39:16 2034 GMT
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIF1TCCBL2gAwIBAgIQCB57XD7Qur4Kdu2XZaLK+DANBgkqhkiG9w0BAQsFADA8
MQswCQYDVQQGEwJVUzEPMA0GA1UEChMGQW1hem9uMRwwGgYDVQQDExNBbWF6b24g
UlNBIDIwNDggTTAxMB4XDTIzMDMzMTAwMDAwMFoXDTI0MDQyODIzNTk1OVowFzEV
MBMGA1UEAxMMbWVkaWEuZ292Lmt3MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIB
CgKCAQEA51qx8Qng5LYZ6RbgOV+40gqnbj1uDafdVwNEHSb65IqYPf0H7RHjB+hb
jaXtSaDxmjUU0E+L/G5TVCy8Vj/D25twICS8TWxwN4nEklHWVsoOuo+oN0cvpfi1
L4H4n6+tUE8enHZ7JakOij5wmfi21FpVuMsdFrjXBFMe2cLYiSPMQg0abjWbieOA
8hu+i/JE1hU6nMwLSEDxrJzamD+B6Edf9gPmKX+RnEw4tn83+GxeV45C7Tj16FBI
rDjqhcwiXSyVqaeUGR8qrSDIxmZLISuGp/B+kVa+xJNx7Z2Tnz/4ZKvuNgHYdwuA
elXEJmPX0xUulhme75kx3yHoC9vTJwIDAQABo4IC9jCCAvIwHwYDVR0jBBgwFoAU
gbgOY4qJEhjl+js7UJWf5uWQE4UwHQYDVR0OBBYEFMXsDM5WXM8O1M5I2GKfYlek
OGO3MCcGA1UdEQQgMB6CDG1lZGlhLmdvdi5rd4IOKi5tZWRpYS5nb3Yua3cwDgYD
VR0PAQH/BAQDAgWgMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjA7BgNV
HR8ENDAyMDCgLqAshipodHRwOi8vY3JsLnIybTAxLmFtYXpvbnRydXN0LmNvbS9y
Mm0wMS5jcmwwEwYDVR0gBAwwCjAIBgZngQwBAgEwdQYIKwYBBQUHAQEEaTBnMC0G
CCsGAQUFBzABhiFodHRwOi8vb2NzcC5yMm0wMS5hbWF6b250cnVzdC5jb20wNgYI
KwYBBQUHMAKGKmh0dHA6Ly9jcnQucjJtMDEuYW1hem9udHJ1c3QuY29tL3IybTAx
LmNlcjAMBgNVHRMBAf8EAjAAMIIBfwYKKwYBBAHWeQIEAgSCAW8EggFrAWkAdwDu
zdBk1dsazsVct520zROiModGfLzs3sNRSFlGcR+1mwAAAYc1xZrCAAAEAwBIMEYC
IQCUiv1HUJZ+nUv4kKB5PjsJ1jcKXypyTHZck7QDxiXmCQIhAKBXnVV6MbL6oVdB
zE+PQd5eGXTGhiR33MSaOm9BllimAHcAc9meiRtMlnigIH1HneayxhzQUV5xGSqM
a4AQesF3crUAAAGHNcWbDQAABAMASDBGAiEAkgilSO+yttUQrMujkf5g62ABpHVA
YZyRr3qfpCEeRJ4CIQDbpm9ZEA7+/yYGgQgPVj+UNEuk3T5gY+3p1Sg8Sk6SrQB1
AEiw42vapkc0D+VqAvqdMOscUgHLVt0sgdm7v6s52IRzAAABhzXFmtoAAAQDAEYw
RAIgU1WqGtX4hfsj6ng9vI4y1jPqjlxcVXoN1O/BSRC51oICIFKvV0fN4ObXzoa8
Jepk5hgtqaQInqoj5/YFcUd9docyMA0GCSqGSIb3DQEBCwUAA4IBAQDKYiYZMrhG
7wSXoZ5Sf+iwIcWa6gTC/v8pgqjS8uclZwf4S4eXKVKQ6+3q3z4Va/4sWSLMCjg/
DKbO52cMF6cP5Ep8370TZl8e7SBqQRBAtCqPRQadiiUVODGIP/2KEhKU50XJd48p
GssCGmgsRP3ae5jHR5K6zsWO65a8ek1SzsZBGd7ruv0da6q+zLSFCQ+Xob7s1AU4
4FUyHZGF6b3Ggs+apReUbxG1tXSJY2cHM/z8VM2s71Zwnf6wyaPnEC6N+loNmyE+
lwoKMEs/GTseNmm7/lG/XBCPOetcUuVXBD7zr9t9O3JASAw/QgPAwr9y3r74/VUu
WWhfIDzn7U+E
-----END CERTIFICATE-----
subject=CN = media.gov.kw
issuer=C = US, O = Amazon, CN = Amazon RSA 2048 M01
---
No client certificate CA names sent
Peer signing digest: SHA256
Peer signature type: RSA-PSS
Server Temp Key: X25519, 253 bits
---
SSL handshake has read 5497 bytes and written 378 bytes
Verification: OK
---
New, TLSv1.3, Cipher is TLS_AES_128_GCM_SHA256
Server public key is 2048 bit
This TLS version forbids renegotiation.
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---
DONE
```

```
$ echo | openssl s_client -verifyCAfile ~/venv/streamlink-311/lib/python3.11/site-packages/certifi/cacert.pem -connect kwmedia.hibridcdn.net:443
CONNECTED(00000003)
depth=2 C = US, ST = New Jersey, L = Jersey City, O = The USERTRUST Network, CN = USERTrust RSA Certification Authority
verify return:1
depth=1 C = GB, ST = Greater Manchester, L = Salford, O = Sectigo Limited, CN = Sectigo RSA Domain Validation Secure Server CA
verify return:1
depth=0 CN = *.hibridcdn.net
verify return:1
---
Certificate chain
 0 s:CN = *.hibridcdn.net
   i:C = GB, ST = Greater Manchester, L = Salford, O = Sectigo Limited, CN = Sectigo RSA Domain Validation Secure Server CA
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Sep  7 00:00:00 2023 GMT; NotAfter: Oct  6 23:59:59 2024 GMT
 1 s:C = GB, ST = Greater Manchester, L = Salford, O = Sectigo Limited, CN = Sectigo RSA Domain Validation Secure Server CA
   i:C = US, ST = New Jersey, L = Jersey City, O = The USERTRUST Network, CN = USERTrust RSA Certification Authority
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA384
   v:NotBefore: Nov  2 00:00:00 2018 GMT; NotAfter: Dec 31 23:59:59 2030 GMT
 2 s:C = US, ST = New Jersey, L = Jersey City, O = The USERTRUST Network, CN = USERTrust RSA Certification Authority
   i:C = GB, ST = Greater Manchester, L = Salford, O = Comodo CA Limited, CN = AAA Certificate Services
   a:PKEY: rsaEncryption, 4096 (bit); sigalg: RSA-SHA384
   v:NotBefore: Mar 12 00:00:00 2019 GMT; NotAfter: Dec 31 23:59:59 2028 GMT
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIGNTCCBR2gAwIBAgIRAMgYg9UzBOdGmuZNBBgJbnEwDQYJKoZIhvcNAQELBQAw
gY8xCzAJBgNVBAYTAkdCMRswGQYDVQQIExJHcmVhdGVyIE1hbmNoZXN0ZXIxEDAO
BgNVBAcTB1NhbGZvcmQxGDAWBgNVBAoTD1NlY3RpZ28gTGltaXRlZDE3MDUGA1UE
AxMuU2VjdGlnbyBSU0EgRG9tYWluIFZhbGlkYXRpb24gU2VjdXJlIFNlcnZlciBD
QTAeFw0yMzA5MDcwMDAwMDBaFw0yNDEwMDYyMzU5NTlaMBoxGDAWBgNVBAMMDyou
aGlicmlkY2RuLm5ldDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAMJt
1ZdHKVv/15Yj6x88O/w1KqQhQoPFoXBl3hjk2xUCaIv3ZpwwZ1/eQmzqfyazJqxQ
NClsDBiPHVYiCRKYNqEQfDqqJyInD2iXCrhhkZUKL5nBIpOcIl5xiB6LBqZsfK1H
+SpoQJV4x/29+YENJzGmzNlBh0pIm2Ix7BGTPQXKblndrMLLn7dXItUT/0PoXxpK
4Nz8iZvRDv1vVVdxjrfsaanCcoS5rXHvv3jJaDOnnUJllSAryd2d/NoemiNumyvh
JkpvapSnezOyIdi1G/dwMyH6SFbuo/bi4DqBf1Fslo5ZxKwFgZR2YtSydIBw9sZi
X5YVs+P4v/D7iCQSJC0CAwEAAaOCAv4wggL6MB8GA1UdIwQYMBaAFI2MXsRUrYrh
d+mb+ZsF4bgBjWHhMB0GA1UdDgQWBBRmrojRuBhShKr7lVKCkhxsZ11HyjAOBgNV
HQ8BAf8EBAMCBaAwDAYDVR0TAQH/BAIwADAdBgNVHSUEFjAUBggrBgEFBQcDAQYI
KwYBBQUHAwIwSQYDVR0gBEIwQDA0BgsrBgEEAbIxAQICBzAlMCMGCCsGAQUFBwIB
FhdodHRwczovL3NlY3RpZ28uY29tL0NQUzAIBgZngQwBAgEwgYQGCCsGAQUFBwEB
BHgwdjBPBggrBgEFBQcwAoZDaHR0cDovL2NydC5zZWN0aWdvLmNvbS9TZWN0aWdv
UlNBRG9tYWluVmFsaWRhdGlvblNlY3VyZVNlcnZlckNBLmNydDAjBggrBgEFBQcw
AYYXaHR0cDovL29jc3Auc2VjdGlnby5jb20wKQYDVR0RBCIwIIIPKi5oaWJyaWRj
ZG4ubmV0gg1oaWJyaWRjZG4ubmV0MIIBfAYKKwYBBAHWeQIEAgSCAWwEggFoAWYA
dQB2/4g/Crb7lVHCYcz1h7o0tKTNuyncaEIKn+ZnTFo6dAAAAYpvlm6hAAAEAwBG
MEQCIA/vCpkX+jd9KjMp9XNM2d80jgJCMThJwG8RqDvOH4GfAiAHDqbm4MIOugvh
l2iYSot3Ftn6XuAthkqtCRMH7fGUdAB1ANq2v2s/tbYin5vCu1xr6HCRcWy7UYSF
NL2kPTBI1/urAAABim+Wbu8AAAQDAEYwRAIgYvKmuwRHNTTATKwDOcFyyiAptljw
hJIy6mr4Og09NN4CIDGffYLAkLlzAeypkTy1/zzrDmkcXVuxqusfaeFvg7IuAHYA
7s3QZNXbGs7FXLedtM0TojKHRny87N7DUUhZRnEftZsAAAGKb5ZvHAAABAMARzBF
AiEA7R4VAnCdO8UalaBPQHihITnSvTMVZ45QGWQupqtVlTwCIFEQyj/QdimqQFH7
i8nq0YZkVfVsRzLh/JgNuldwaWgeMA0GCSqGSIb3DQEBCwUAA4IBAQDJXvSVNXq8
KX6O+Ldvw8OsZ5zMeUVEhvITmv364fcNKhGC/QTmRSI3Vdinwc2+MlAFFhHgy6Vc
85FGPI8BzTNtIaJ/UQ7QOLc/8CldHCVmgsEGOQDPxwfAzoSO/lyTYImh+XdlyVfQ
XYbsjUFzuiXsWAPpECMgvjvep+b1w0hZdvHTWoHeR6p8Z2b4nD4jWDlWe031zJoU
RXWnBghTTjzS+K1WJDxhPUsR7pTu9Qc/1yqzJVFw+S+3+Sy81oZWLl/T7yCeQsx9
+nnSsgEOxbBZr2EyQLh52kDEc5gOh2VlCEu3SDbq61hTFH+rvkHEQWLgCJOEEqid
PqZD4Rn1s0Q7
-----END CERTIFICATE-----
subject=CN = *.hibridcdn.net
issuer=C = GB, ST = Greater Manchester, L = Salford, O = Sectigo Limited, CN = Sectigo RSA Domain Validation Secure Server CA
---
No client certificate CA names sent
Peer signing digest: SHA256
Peer signature type: RSA-PSS
Server Temp Key: X25519, 253 bits
---
SSL handshake has read 5135 bytes and written 403 bytes
Verification: OK
---
New, TLSv1.3, Cipher is TLS_AES_256_GCM_SHA384
Server public key is 2048 bit
This TLS version forbids renegotiation.
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---
DONE
```

**Reply by andaluspc** [reply]
Thank you Sebastian for taking the time to respond. I checked manually previously. the cert seems valid in terms of date (root CA is already trusted in my machine).
However, the complain is about hostname of server doesn't match the subject of the certificate. (I've taken some screenshots on a different machine that I can share later if necessary).

The cause seems to be geographical related!! They seem to have multiple servers and one of the servers have bad subject in certificate. 
In output you put it shows:
subject=CN = media.gov.kw
However, the subject in the certificate (when it fails) shows some wildcard to different URL (*.llhs.xyz.xyz). I don't remember exact URL.

seems if I am redirected to a hosting server with bad CN in certificate I got the issue. If I am redirected to a good server, the issue doesn't present.

I need the stream URL because I pull it to one m3u8 on my home server (for which I use phone and TV to watch channels in that m3u8 file). The channel is valid for 24 hours so I have to update channels frequently via a script to update m3u8 with new stream URL!

I thought at the beginning the issue can be server blocked my IP because I had to try my script to pull channels multiple times while testing. However, I could sometimes change my public IP and it didn't get issue to work.

**Reply by bastimeyer** [reply]
> They seem to have multiple servers and one of the servers have bad subject in certificate.

Then your problem is DNS based.
Use a different DNS server in order to resolve the hostname to a specific IP address, or manually set it to a known good one.

> I need the stream URL

https://streamlink.github.io/cli.html#cmdoption-stream-url

**Reply by andaluspc** [reply]
Thanks Sebastian. I usually use my home DNS (pi-hole) but even if I try public DNS (8.8.8.8, 9.9.9.9, 1.1.1.1) all show failure scenario.
The issue isn't DNS based as the redirection (if my guessing is correct) happens from customer's side (like a load balancer loads traffic to multiple physical or virtual servers)

Today morning channels were working fine. Now they don't:

```
Desktop:~$streamlink https://www.media.gov.kw/LiveTV.aspx?PanChannel=KTV1 --stream-url
error: Unable to open URL: https://hiplayer.hibridcdn.net/l/kwmedia-kwtv1 (HTTPSConnectionPool(host='hiplayer.hibridcdn.net', port=443): Max retries exceeded with url: /l/kwmedia-kwtv1 (Caused by SSLError(SSLCertVerificationError(1, "[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: Hostname mismatch, certificate is not valid for 'hiplayer.hibridcdn.net'. (_ssl.c:1007)"))))
```

without ssl check it works (however, stream doesn't work on my IPTV player. only works on vlc with many cert warning pop-ups)

```
DESKTOP:~$ streamlink https://www.media.gov.kw/LiveTV.aspx?PanChannel=KTV1 --stream-url --http-no-ssl-verify
https://hiplayer.hibridcdn.net/p/kwmedia-kwtv1?p=48&s=1695835704&e=1695922104&cf=1695922104&h=58cf7cb3535be1999616d12d1c61b464
```


![image](https://github.com/streamlink/streamlink/assets/134834167/afdf772e-d253-4c83-bdda-c68b656a76bb)



```
@DESKTOP-O37QKDH:~$ openssl s_client -verify_hostname -connect hiplayer.hibridcdn.net:443
CONNECTED(00000003)
depth=0 CN = *.hs.llnwd.net
verify error:num=62:hostname mismatch <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
verify return:1
depth=2 C = US, ST = New Jersey, L = Jersey City, O = The USERTRUST Network, CN = USERTrust RSA Certification Authority
verify return:1
depth=1 C = GB, ST = Greater Manchester, L = Salford, O = Sectigo Limited, CN = Sectigo RSA Domain Validation Secure Server CA
verify return:1
depth=0 CN = *.hs.llnwd.net
verify return:1
<truncated output>
```

**Comment by andaluspc** [comment]
I just did quick check. The hostname I got redirected to is different for working and non-working scenarios. I've got different certs with same on the browser.

in working scenario:
![image](https://github.com/streamlink/streamlink/assets/134834167/7d3d5d60-b2e0-4de8-8386-5df7a1fc39c5)



in NON-WORKING scenario:

![image](https://github.com/streamlink/streamlink/assets/134834167/5b85f449-9033-48b9-b2a3-3f83b0263072)

I am confused why it returns different certs!! this is what I mean when I say it redirects to different hosts and end up having different certs (not sure if this is done from server side by a device in the middle like firewall or something)

**Comment by andaluspc** [comment]
Obviously, as you mentioned Sebastian, issue is related to DNS! The IP address for the host get changed over time:

Working scenario:
```
DESKTOP:~$ nslookup hiplayer.hibridcdn.net
Server:         8.8.8.8
Address:        8.8.8.8#53
Non-authoritative answer:
hiplayer.hibridcdn.net  canonical name = hibrid.s.llnwi.net.
Name:   hibrid.s.llnwi.net
Address: 178.79.208.44 <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
```
Non-Working Scenario
```
DESKTOP:~$ nslookup hiplayer.hibridcdn.net
Server:         8.8.8.8
Address:        8.8.8.8#53

Non-authoritative answer:
hiplayer.hibridcdn.net  canonical name = hibrid.s.llnwi.net.
Name:   hibrid.s.llnwi.net
Address: 91.106.111.49 <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
```

Will do more tests to DNS thing. I currently fixed a working IP in hosts file and will see if this will get issue resolved.

**Comment by andaluspc** [comment, marked as answer]
Solved after some investigation.
I added hostname  hiplayer.hibridcdn.net to hosts files with ip 178.79.208.44.  then streamlink could get stream-url without problem. 
However, the video and stream doesn't play! (in web page of https://www.media.gov.kw/LiveTV.aspx?PanChannel=KTV1, it was not showing the player before, now it is showing the player but the player doesn't play the stream).
With further investigation I could find there is another domain need to be fixed: kwmedia.hibridcdn.net. This domain was resolving to a different IP then hiplayer.hibridcdn.net (in working scenario, they resolved to same IP. but in my scenario at this stage they were resolving to different IPs because one resolves from hosts file and one from DNS).
I tested with fixing both hosts to same IP in hosts file and it worked. I eventually fixed both hosts to same IP address in my local DNS and pointed all machines I use to local DNS then public DNS. now everything seems to work as supposed.

Thanks Sebastian for your kind assistance.
